pinot-core/src/main/java/org/apache/pinot/core/common/DataFetcher.java

```diff
@@ -285,7 +285,7 @@ void readIntValues(int[] docIds, int length, int[] valueBuffer) {
         _reader.readDictIds(docIds, length, dictIdBuffer, readerContext);
         _dictionary.readIntValues(dictIdBuffer, length, valueBuffer);
       } else {
-        switch (_reader.getValueType()) {
+        switch (_reader.getValueType().getStoredType()) {
```

Comment:
We don't need stored type lookup here as the value type of reader is already stored type

Reply:
Fixed.